### ISSUE TITLE
Navigation problem 

### ISSUE THREAD

**Issue by Gulshanaggarwal** [NucleoidJS/Nucleoid#17]
**What is the issue?**
Navigation on the  [Learn](https://nucleoid.com/learn) page of NucleoidJS doesn't work properly.

**How to reproduce?**

- Goto the [Learn](https://nucleoid.com/learn) page  
- Click on Express.js in the right-side navigation bar it doesn't scroll down to that section. Navigation doesn't work properly for all learning sections.

**Comment by Gulshanaggarwal** [comment]
@canmingir  can I start working on the issue?

**Comment by canmingir** [comment]
👍 Sounds good, honestly, I couldn't find a good ToC library for React. BTW, HTML is generated thru markdown [here](https://github.com/NucleoidJS/nucleoid.com/blob/203f093be6cb5a9780d97ad284d294c05e2688eb/src/pages/Learn.js#L49)

**Comment by Gulshanaggarwal** [comment]
@canmingir I have figured out why it's not working. If you clearly give a look at hyperlinks(on contents) that have either **`?`**  or **`,`** or **`.js`** included is not working. 

**Let's see through the image below -**
 Here I removed  **`?`** from What is Nucleoid   markdown  and hyperlink is working now.

![Screenshot (312)](https://user-images.githubusercontent.com/58553401/169643103-98f1b32d-5acf-4724-b6d6-3d66e1f8de6c.png)


**Should I fix it by removing like this or something else?** 🤔

**Comment by Gulshanaggarwal** [comment]
@canmingir what do you think about the previous comment ?

**Comment by canmingir** [comment]
Yes, you are right, the question mark is causing an issue in [here](https://github.com/NucleoidJS/nucleoid.com/blob/04e8dc48ffbc0a3afb2d5f5b0e399da86453fc92/src/pages/Learn.js#L89-L95). Instead of deleting an question mark from the content, let's see there is a workaround in `react-markdown-heading`, which I've opened up an issue on their side. BTW, I am in search of finding better ToC library as well, lmk if you know any of them,

https://github.com/kyoncy/react-markdown-heading/issues/33

**Comment by Gulshanaggarwal** [comment]
@canmingir  I tried to generate the toc with another library [react-toc](https://www.npmjs.com/package/react-toc) but it has the same problem but one thing which I figured out is that organizations like [socket.io](https://github.com/socketio/socket.io-website) or [enarx](https://github.com/enarx/website) uses [docusaurus](https://docusaurus.io/) tool with markdown files to generate their website content including toc.

**What do you think** 🤔

**Comment by canmingir** [comment]
Fantastic, [docusaurus](https://docusaurus.io/) is looking really cool, we can switch into that, instead of trying to maintain our markdown generator

**Comment by Gulshanaggarwal** [comment]
@canmingir So, what's the plan for switching? Do you have any idea so that we can start?

**Comment by canmingir** [comment]
I would say yes, let's switch, do you want to give a shot?

**Comment by Gulshanaggarwal** [comment]
@canmingir in open-source always! let's give it a shot.  What's the plan?

**Comment by canmingir** [comment]
I've been looking at it, looks like it won't be easy switch over, but I think this gives better handling on our docs, I will be working on it, but it may take awhile tho

**Comment by Gulshanaggarwal** [comment]
@canmingir ok, then go ahead. Once you start I will join you.

**Comment by canmingir** [comment]
There is a conversation going on, docusaurus may not be possible with MUI

https://github.com/facebook/docusaurus/issues/7505

**Comment by Gulshanaggarwal** [comment]
@canmingir If MUI is not possible then we can go ahead with what docusaurus provides.

**Comment by canmingir** [comment]
Working on a hybrid solution, I am trying to keep the main page with MUI, but rebuilding rest of pages with docusaurus (They are already in markdown anyway). Let see, how far I can go with

https://github.com/NucleoidJS/nucleoid.com/tree/docusaurus

**Comment by Gulshanaggarwal** [comment]
👍

**Comment by canmingir** [comment]
Finally, docusaurus version:

https://test3.nucleoid.com/

**Comment by Gulshanaggarwal** [comment]
Improved UI great! although there are still problems.

**Comment by Gulshanaggarwal** [comment]
Hey Can, can you tell me if any help I can do?
